ksql-engine/src/main/java/io/confluent/ksql/KsqlEngine.java

```diff
@@ -312,6 +312,66 @@ public QueryMetadata getQueryExecutionPlan(final Query query, final KsqlConfig k
     }
   }
 
+  public List<Pair<String, Statement>> parseStatements(
+      final String queriesString,
+      final MetaStore tempMetaStore
+  ) {
+    final MetaStore tempMetaStoreForParser = tempMetaStore.clone();
+    final KsqlParser ksqlParser = new KsqlParser();
+    final List<SqlBaseParser.SingleStatementContext> parsedStatements;
+    try {
+      parsedStatements = ksqlParser.getStatements(queriesString);
```

Comment:
`ksqlParser.getStatements` already throws a `ParseFailedException`, so catching this and wrapping it in _another_ `ParseFailedException` doesn't seem to add much.

What would add benefit might be to add the SQL string to the exception thrown by the parser.

Reply:
Added the SQL string to the message.

Reply:
So can we get rid of the rewrapping?

Reply:
Yes, removed the extra rewrapping.